homeassistant/components/drop_connect/coordinator.py

```diff
@@ -0,0 +1,184 @@
+"""DROP device data update coordinator object."""
+from __future__ import annotations
+
+import logging
+from typing import Any
+
+from homeassistant.config_entries import ConfigEntry
+from homeassistant.core import HomeAssistant
+from homeassistant.helpers.device_registry import DeviceInfo
+from homeassistant.helpers.update_coordinator import DataUpdateCoordinator
+from homeassistant.util.json import JSON_DECODE_EXCEPTIONS, json_loads
+
+from .const import (
+    CONF_DATA_TOPIC,
+    CONF_DEVICE_DESC,
+    CONF_DEVICE_ID,
+    CONF_DEVICE_NAME,
+    CONF_DEVICE_OWNER_ID,
+    CONF_DEVICE_TYPE,
+    CONF_HUB_ID,
+    DEV_HUB,
+    DOMAIN,
+    KEY_STATUS,
+)
+
+_LOGGER = logging.getLogger(__name__)
+
+
+class DROPDeviceDataUpdateCoordinator(DataUpdateCoordinator):
+    """DROP device object."""
+
+    config_entry: ConfigEntry
+
+    def __init__(self, hass: HomeAssistant, config_entry: ConfigEntry) -> None:
+        """Initialize the device."""
+        self._model: str = f"{config_entry.data[CONF_DEVICE_DESC]} on hub {config_entry.data[CONF_HUB_ID]}"
+        if config_entry.data[CONF_DEVICE_TYPE] == DEV_HUB:
+            self._model = f"Hub {config_entry.data[CONF_HUB_ID]}"
+        self._manufacturer: str = "Chandler Systems, Inc."
+        self._device_name: str = config_entry.data[CONF_DEVICE_NAME]
+        self._device_information: dict[str, Any] = {}
+        super().__init__(hass, _LOGGER, name=f"{DOMAIN}-{config_entry.unique_id}")
```

Comment:
You can move the `super().__init__()`. In that initializer, it will set `self.config_entry`. This way you can remove that parameter from this constructor

Reply:
I'm not following...  Is the goal to remove the `config_entry` class variable?  If so, removing it causes `mypy` to emit errors saying `Item "None" of "ConfigEntry | None" has no attribute "data"` in the `drop_message_received()` callback and `device_info()` property.  Sorry for the extra attention you're having to apply to this PR; I'm an embedded C developer and this is actually my first Python project...

Reply:
Not the config_entry class variable but the config entry constructor parameter

Reply:
The coordinator constructor will automatically assign the config_entry class variable with the right config_entry. (But since you can use a coordinator without config entry, it can be None. Hence we need to overwrite it

Reply:
Got it. I was overcomplicating it. It's been fixed in abac0cf9b286522a87dda7e33e0c80bafc6ebf1e.